src/ledger/LedgerStateTests.cpp

```diff
@@ -1208,22 +1208,20 @@ testAllOffers(
         {
             REQUIRE(expectedIter->first == iter->first);
 
-            auto const& expectedInner = expectedIter->second;
+            auto expectedInner = expectedIter->second;
             auto const& inner = iter->second;
-            auto expectedInnerIter = expectedInner.cbegin();
-            auto innerIter = inner.cbegin();
-            while (expectedInnerIter != expectedInner.end() &&
-                   innerIter != inner.end())
+
+            REQUIRE(expectedInner.size() == inner.size());
+            for (auto& innerCur : inner)
             {
-                auto const& oe = innerIter->current().data.offer();
-                REQUIRE(*expectedInnerIter ==
-                        std::make_tuple(oe.offerID, oe.buying, oe.selling,
-                                        oe.amount));
-                ++expectedInnerIter;
-                ++innerIter;
+                auto const& oe = innerCur.current().data.offer();
+                auto d = std::make_tuple(oe.offerID, oe.buying, oe.selling,
+                                         oe.amount);
+                auto expectedInnerIter =
+                    std::find(expectedInner.begin(), expectedInner.end(), d);
+                REQUIRE(expectedInnerIter != expectedInner.end());
+                expectedInner.erase(expectedInnerIter);
```

Comment:
Why call `erase` here? Also, if you didn't erase here then there would be no need to copy-assign `expectedInner`.

Reply:
yes, first version I had didn't do this, but I thought the test code could have bad data and we would not catch it. This guarantees that we don't do stupid things like having the same data specified several times in the expected set as it can only be used once.